src/GatewayRegistry.js

```diff
@@ -7,13 +7,19 @@ export default class GatewayRegistry {
     this._currentId = 0;
   }
 
+  _getChildIndex(childId) {
+    return +childId.match(/\d+$/)[0]
+  }
+
   _renderContainer(name) {
     if (!this._containers[name] || !this._children[name]) {
       return;
     }
 
     this._containers[name].setState({
-      children: Object.keys(this._children[name]).sort().map(id => this._children[name][id])
+      children: Object.keys(this._children[name])
+          .sort((a, b) => this._getChildIndex(a) - this._getChildIndex(b))
+          .map(id => this._children[name][id])
```

Comment:
@kseniya57 do you know if the `name` even needs to be part of the key; i.e. any reason ln 47 can't just be:

```js
const gatewayId = `this._currentId`;
```

That way this could avoid the regex and instead cast it to a number?

Reply:
I added 'name' just to have a more verbose id. 

I'm not sure I'd want to remove `name` in ln 47 just to have this usable for sorting. 
I'm thinking I'd rather introduce a new prop and have it used in sorting instead of relying in `gatewayId`